src/utils.jl

```diff
@@ -246,3 +276,201 @@ function _slack_dual_vectors(model)
     end
     return (s, y)
 end
+
+
+# extend caching optimizer
+function MOI.set(
+    m::MOI.Utilities.CachingOptimizer,
+    attr::AbstractDiffAttribute,
+    index::MOI.Index,
+    value,
+)
+    if m.state == MOI.Utilities.ATTACHED_OPTIMIZER
+        optimizer_index = m.model_to_optimizer_map[index]
+        optimizer_value = MOI.Utilities.map_indices(m.model_to_optimizer_map, value)
+        # if m.mode == AUTOMATIC
```

Comment:
should this be removed? Also should we add a docstring to justify this method?

Reply:
This was copied (and simplified) from MOI´s caching optimizer´s code. (https://github.com/jump-dev/MathOptInterface.jl/blob/f585f81caddd54b8cc9843c263ee89b28ebc9a9e/src/Utilities/cachingoptimizer.jl#L662)
We need this to allow `AbstractDiffAttribute` to flow through this layer.
Referencing MOI's code in a comment is a good idea.

We might want add back the `if m.mode == AUTOMATIC` block.

I think we can stick with the `ATTACHED_OPTIMIZER` error for simplicity.